### PR TITLE
feat: sign delegate with ununifi & keplr

### PR DESCRIPTION
**Pull request by Senna46**
Send createDelegate Tx with Keplr or UnUniFi

- [x] Keplr
keplr

- [x] UnUniFi (Legacy)
legacy

### CODE REVIEW

File: projects/portal/src/app/models/cosmos/tx-common.service.ts
Comment (YasunoriMATSUOKA) [inline projects/portal/src/app/models/cosmos/tx-common.service.ts:140]: I guess it is better to remove this function (or use this function to execute keplrAppService.signDirect in this function.)

```suggestion
```

**Comment by Senna46** [comment]
@mkXultra @taikiFurusyo 
Use https://github.com/UnUniFi/web-apps/tree/feature/delegate branch to check the behavior of Legacy and Keplr.
Now, Available only for Delagate.

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
@Senna46 
I commented on what I felt about this PR diff.
I would like to get a review from @mkXultra before implement other similar features.

File: projects/portal/src/app/models/cosmos/tx-common.service.ts
Comment (Senna46) [reply projects/portal/src/app/models/cosmos/tx-common.service.ts]: Changed! thx

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
@Senna46 
The following methods are not this PR diff, but it may be better to move the wallet related code from keplr.application.service to walletApplicationService.

- keplrSelectWallet
- openKeplrImportWalletDialog ... I guess openKeplrSelectWalletDialog is better.
- openConnectWalletCompleteDialog ... I guess you don't need implement this because same methods has already exists on WalletApplicationService.

**Comment by Senna46** [comment]
> @Senna46 The following methods are not this PR diff, but it may be better to move the wallet dialog related code from keplr.application.service to walletApplicationService.
> 
> * keplrSelectWallet
> * openKeplrImportWalletDialog ... I guess openKeplrSelectWalletDialog is better.
> * openConnectWalletCompleteDialog ... I guess you don't need implement this because same methods has already exists on WalletApplicationService.

Reconsider after merge as it will not affect UX and will complicate this PR.

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
@Senna46 
I missed another important point.

I guess now you are using cosmojs/launchpad. But it seems to support only  Cosmos SDK 0.37 (cosmoshub-3), 0.38 and 0.39 (Launchpad). ( 
by https://www.npmjs.com/package/@cosmjs/launchpad/v/0.24.0-alpha.3 )

I guess we should use cosmojs/stargate( https://www.npmjs.com/package/@cosmjs/stargate ). how do you think about it?

**Comment by YasunoriMATSUOKA** [comment]
> @YasunoriMATSUOKA
> I am using this only for Decode and don't see any problem. Probably decode in base64.
> 
> I have not found a similar feature in stargate.
> https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/signingstargateclient.ts

If it's just base64 decoding, can you implement it without using cosmojs and remove cosmojs from package.json?
I would like to eliminate dependencies on libraries that only support older versions as soon as possible.

**Comment by YasunoriMATSUOKA** [comment]
@Senna46 
I guess we should set `chainName` in `config.js` and use it in `suggestChain`.

examples

- UnUniFi
- UnUniFi (Beta)
- UnUniFi (Alpha)

image

**Review by mkXultra — APPROVED** [review]

**Pull request merged**
